[DO NOT MERGE] Apollos 2.0.0 [BUILD PREVIEW]

conrad-vanl opened the pull request.

Lepozepo committed:
Add bcrypt and meteor-random

Lepozepo committed:
Add bcrypt and meteor-random

Lepozepo committed:
Start buliding resolvers for mutations

Lepozepo committed:
Start buliding resolvers for mutations

Lepozepo committed:
Finish deauthorizeUser

Lepozepo committed:
Finish deauthorizeUser

Lepozepo committed:
Simplify responses

Lepozepo committed:
Simplify responses

Lepozepo committed:
Add graphql-date

Lepozepo committed:
Add graphql-date

Lepozepo committed:
Add custom Date scalar

Lepozepo committed:
Add custom Date scalar

Lepozepo committed:
Add liquid-node

Lepozepo committed:
Add liquid-node

Lepozepo committed:
Use basic auth instead

Lepozepo committed:
Use basic auth instead

Lepozepo committed:
Start working with email

Lepozepo committed:
Start working with email

Lepozepo committed:
Add pascal case and snake case

Lepozepo committed:
Add pascal case and snake case

Lepozepo committed:
Add GUID function

Lepozepo committed:
Add GUID function

Lepozepo committed:
Bring in sendEmail function

Lepozepo committed:
Bring in sendEmail function

Lepozepo committed:
Rename email

Lepozepo committed:
Rename email

Lepozepo committed:
Use sendEmail

Lepozepo committed:
Use sendEmail

Lepozepo committed:
Install axios to simplify API

Lepozepo committed:
Install axios to simplify API

Lepozepo committed:
Create axios instance

Not sure about the headers on this one

Lepozepo committed:
Create axios instance

Not sure about the headers on this one

Lepozepo committed:
Integrate with API

Lepozepo committed:
Integrate with API

Lepozepo committed:
Restore current token functionality

Just in case

Lepozepo committed:
Restore current token functionality

Just in case

Lepozepo committed:
Silly logic

Lepozepo committed:
Silly logic

Lepozepo committed:
Build forgotPassword, resetPassword, and changePassword functions

Lepozepo committed:
Build forgotPassword, resetPassword, and changePassword functions

Lepozepo committed:
Register mutation resolvers and responses

Lepozepo committed:
Register mutation resolvers and responses

Lepozepo committed:
Add ROCK_TOKEN_NAME to .env.example

Lepozepo committed:
Add ROCK_TOKEN_NAME to .env.example

Lepozepo committed:
Merge branch '107a-user-auth' of https://github.com/Differential/Heighliner into 107a-user-auth

Lepozepo committed:
Wire up new apollos mutations

Lepozepo committed:
Patch up response objects

Lepozepo committed:
Hard-code ROCK_TOKEN_NAME

It should be consistent across rocks

conrad-vanl committed:
Add video prop to content

conrad-vanl committed:
Update playerConfig url

conrad-vanl committed:
fix typo in string, return null if no video

Lepozepo committed:
Patch baseURL

Lepozepo committed:
Login via props instead of via cookie context

Lepozepo committed:
Remove axios

Lepozepo committed:
Rebuild API

Lepozepo committed:
Use rebuilt api

Lepozepo committed:
Patch user schema with backwards compatibility

Lepozepo committed:
Update person resolver

Lepozepo committed:
Attach person to context

Lepozepo committed:
Add some degree of encryption and prepare for tokenization flow

Lepozepo committed:
Respond with the response even if it isn't JSONable

Lepozepo committed:
Add support for indexes

Lepozepo committed:
Make sure mongo models use global.Promise

Lepozepo committed:
Get forgot password working

Lepozepo committed:
Make changeUserPassword work

Lepozepo committed:
Patch sendEmail URLs

Lepozepo committed:
Get sendEmail working

The emails received aren't very good but maybe that's how it's supposed to be?

Lepozepo committed:
Update model to use sendEmail and fix some error messages

Lepozepo committed:
Patch "connect should return status"

A status can be true or false

Lepozepo committed:
It should return the real ID

Is there a specific reason for encryption here?

Lepozepo committed:
Add simple auth tokens

Now we're secure

Lepozepo committed:
Clean up and add logout

Lepozepo committed:
Patch logout

Lepozepo committed:
Remove dev config

Lepozepo committed:
Merge pull request #332 from Differential/107a-user-auth

107a user auth

conrad-vanl committed:
Merge pull request #335 from Differential/migrate-ooyalaid-to-video-prop

Add video prop to content

Lepozepo committed:
Patch person context errors

Lepozepo committed:
Patch responses to use PrimaryAliasId as Id

Lepozepo committed:
Patch tests

Lepozepo committed:
Merge pull request #341 from NewSpring/190a-user-ids

190a user ids

Lepozepo committed:
Use PrimaryAliasId to fetch liked content

Lepozepo committed:
Pass PrimaryAliasId to toggle likes

Lepozepo committed:
Add hasUserLike to likes model

Lepozepo committed:
Add hasLike resolver

Lepozepo committed:
Patch user offline query

Lepozepo committed:
Patch tests

newspringweb commented:
<!--
  0 failure: 
  2 warning:  Please assign som..., :exclamation: Big PR
  
  1 markdown notices
-->


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>Please assign someone to merge this PR, and optionally include people who should review.</td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>:exclamation: Big PR</td>
    </tr>
  </tbody>
</table>


---

# QA Checklist

- [ ] do something
- [ ] do something else

<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

Lepozepo committed:
User exists hotfix (#344)

* Check if a user exists before attempting to create it

* Remove console.log

Lepozepo committed:
Merge branch 'alpha-2.0.0' into 137a-content-likes

Lepozepo committed:
Rename hasLike to isLiked

Lepozepo committed:
Merge pull request #342 from NewSpring/137a-content-likes

137a content likes

Lepozepo committed:
Move followable topics to GraphQL (#345)

* Add FOLLOWABLE_TOPICS constant

* Build followable topics functions on the user model

* Build schema and mutations

* Build query and mutation resolvers

* Remove need for channels option on the client

* Clean up

* Return all topics on error

* Patch tests

* Add followed topics to current person also

* Use param instead of context

* Add topics query

For the sake of completion

* Remove unused async

* Add some tests

* Also add a test for followable topics

conrad-vanl commented on src/rock/models/feeds/queries.js:
```diff
@@ -4,7 +4,6 @@ export default [
 
   `userFeed(
     filters: [String]!
-    options: String = "{}"
```
@Lepozepo I didn't catch this in the review, but we can't remove arguments like this from the GraphQL schema less we break all existing clients. So we need to add this back in.

See http://graphql.org/learn/best-practices/#versioning

Might be worth looking at something like https://github.com/entria/graphql-find-breaking-changes-cli in the future.

Lepozepo replied:
@conrad-vanl it may be a bit more complex to maintain compatibility considering that not all of the previous clients’ data is coming from GraphQL. We could do it but I think we’ll need to either force update the old client to understand the new DB structure (easy to do because of cordova) or write something to keep the migrated collections in sync with the old structure.

conrad-vanl replied:
> We could do it but I think we’ll need to either force update the old client to understand the new DB structure (easy to do because of cordova) or write something to keep the migrated collections in sync with the old structure.

Expand on this some more. GraphQL is a living api schema, and so breaking changes _cannot_ be implemented. If we have other breaking changes in this branch right now, we need to roll them back.

Fields can be marked `deprecated` and they are removed from documentation, but we still need to support old fields. Query params like this - need to still be a part of the schema. Unfortunately, this attribute (`options`) being a JSON string was probably a not-too-great design decision for graphql, but we need to have some method of supporting it so that old clients aren't hit with errors (force updating old clients isn't a suitable option). Right now if you were to point Holtzman at alpha, it would crash.

conrad-vanl replied:
So, maybe a potential solution could be to keep options in the schema, but it is ignored...I don't like that idea as that is still altering the data existing clients receive, but it won't cause them to crash 🤔

Lepozepo replied:
@conrad-vanl yes, we can add `options` and prevent the crash that way for sure. What the resolver does to resolve doesn't matter anyway so the extra info coming down the pipeline has no impact.

Lepozepo replied:
My concern is that things like `likes` and `following` now exist in different mongo collections. These mongo collections don't have publishers in Holtzman so Holtzman will fail unless we sync with the legacy stuff or update Holtzman.

Lepozepo replied:
But considering Holtzman is a cordova app, it doesn't matter. Old clients will automatically be upgraded to Apollos anyway and use all the new stuff.

conrad-vanl replied:
Also, how exactly did the DB structure change?

Lepozepo replied:
@conrad-vanl following topics don't exist as an array under the users collection (they have their own collection). Likes didn't change much, the `userId` uses the `PrimaryAliasId` now instead of meteor's user collection _id.

Lepozepo replied:
But both are breaking changes for Meteor pub/sub (as all user ID's have effectively changed to `PrimaryAliasId`). Which is why we'll need to run a migration script before deployment.

conrad-vanl replied:
> But both are breaking changes for Meteor pub/sub

Got it.

> But considering Holtzman is a cordova app, it doesn't matter. Old clients will automatically be upgraded to Apollos anyway and use all the new stuff.

Only if we push a new cordova build with Apollo in it, right? I guess it depends on what's local and what's loaded over the wire... are all cordova apps just wrappers pointing to a website, or are there local javascript/html on the device? I can't remember...it's been so long since I've done cordova stuff

conrad-vanl replied:
I think likes/following topics being moved into a collection in Heighliner is fine, we just need to support backwards compatibility with this `options` argument, which is just used to specify topics to follow, which doesn't depend on the db structure of likes/following, since they're just names...right?

Lepozepo replied:
It's just a wrapper but it creates a bridge via javascript that opens access to native functions (like storage, sound, video, etc). I don't think we'll need to do anything special on our end but I can have a quick look.

Lepozepo replied:
Likes in Holtzman go into an array stored in redux and are compared against whatever is rendered. The publisher searches for likes via `userId` which will change to `PrimaryAliasId` after we migrate. This means users logged in via the old authentication mechanism will have no way of knowing which likes belong to them.

Lepozepo replied:
There are a couple of patches I could add though to make it know after the migration is done. Maybe also put the old `userId` in the likes collection

Lepozepo replied:
I'm sure I can get backwards compatibility working but I'm not sure how long it will take

conrad-vanl replied:
@Lepozepo I'm only concerned about GraphQL schema backwards compatibility, TBH. If the old app is broken because it stores likes locally, w/e. As long as the schema guarantees backwards compatibility, I'm happy.

Lepozepo replied:
Ok, I'll restore it as soon as saved payments is done. Unless it's more urgent?

conrad-vanl replied:
> urgent

Nope, not urgent. Sounds good!

Lepozepo committed:
Add savedPayment query

Lepozepo committed:
Return a blank array if fetching fails

Lepozepo committed:
Repair savedPayment resolver

Lepozepo committed:
Update snapshots

Lepozepo committed:
Saved payment methods are stored immediately on completeOrder

So that we may respond with it

Lepozepo committed:
Patch tests

Lepozepo committed:
Improve responses

Lepozepo committed:
Allow saving payment methods while checking out with a frequency

Lepozepo committed:
Remove ReferenceNumber test

It doesn't apply anymore

Lepozepo committed:
Merge pull request #346 from NewSpring/140a-saved-payments

140a saved payments

Lepozepo committed:
Make sure card expiration dates are saved also

Lepozepo committed:
Update formatTransaction snapshot

Lepozepo committed:
Update resolver spec

Lepozepo committed:
Enable test cards and disable broken SecurityKey test

Lepozepo committed:
Add netlify to whitelist

Lepozepo committed:
Merge pull request #347 from NewSpring/164a-give-ui-gaps

164a give ui gaps

conrad-vanl committed:
add embedUrl prop to live video

conrad-vanl committed:
disable playerConfig on video for now

conrad-vanl committed:
use apollos hosted config files

conrad-vanl committed:
change playerConfig to point to apollos

Lepozepo committed:
Merge pull request #348 from NewSpring/embedUrl-live-video

Add embedUrl to live video

Lepozepo committed:
Allow home to be undefined (#350)

Not all users have an address saved to people but all users have a billing address, therefore this should be queried from NMI

Lepozepo committed:
Add more defaults to statement

Lepozepo committed:
264a details views (#352)

* Resolve whether the schedule is active or not

* Clean up

conrad-vanl committed:
add newspring.github to server whitelist

Lepozepo committed:
205 is liked groups (#353)

* Patch hasUserLike to identify any kind of like

* Patch hasUserLike signatures and add to groups

conrad-vanl committed:
fix regex

Lepozepo committed:
287a profile mutations (#355)

* Build updateProfile resolver

* Do not cache context person

If any relationship changes, caching will break those relationships

* updateProfile can also update Campus by entityId

* Caching should be a buy in not a buy out

Caching needs further work as there is currently no established pattern to invalidate cached results

* Build updateHomeAddress mutation

* Use parseGlobalId for campus

* Patch CampusId error

* Revert opt-in cache

* Patch left overs

Lepozepo committed:
260a image uploads (#356)

* Build attachPhotoIdToUser mutation

* Make sure photo resolver returns something even if there is nothing or an error occurs

* Patch test

conrad-vanl committed:
Patch user registration

@Lepozepo tagging you on this in case you want to look at it at some point. User signups currently failing, this appears to fix it. Although to be honest, I’m not sure I’m following the logic here 100%.

conrad-vanl committed:
try reverting back to fix cors requests

conrad-vanl committed:
add CORS for apollos.netlify and newspring.github

conrad-vanl committed:
add videoUrls to videos (#365)

richarddubay committed:
Merge branch 'master' into beta-2-0-0

Lepozepo committed:
303a toggle topic (#368)

* Patch getFromAliasId to pass along PrimaryAliasId

This is how the person context is being set but because we're not using person from context in this particular query, the result of the person is not mutated like it is in context

* Remove person object mutations from context

richarddubay committed:
Merge branch 'alpha-2.0.0' of github.com:NewSpring/Heighliner into beta-2-0-0

conrad-vanl committed:
add support for wistia video on content (#370)

richarddubay committed:
fixed merge conflicts

richarddubay committed:
removed duplicate declaration of defaultCache

richarddubay committed:
added filter back

Lepozepo committed:
Default people to array

As was in Holtzman, but this is useless anyway

Lepozepo committed:
Comment out broken (and unnecessary) query

Lepozepo committed:
Use limit/offset and remove attributes

You need attributes to order by it without triggering an error

Lepozepo committed:
Patch already broken test

Lepozepo committed:
Update snapshots

Lepozepo committed:
Disable invalid test

Commit:
Merge pull request #373 from NewSpring/374a-ordered-transactions

374a ordered transactions

IsaacHardy committed:
Merge branch 'alpha-2.0.0' of https://github.com/NewSpring/Heighliner into beta-2-0-0

IsaacHardy committed:
udpated nmi key and gateway to env variables

IsaacHardy committed:
removed logs

Commit:
Merge pull request #375 from NewSpring/nmi

udpated nmi key and gateway to env variables

richarddubay committed:
fix the dashboard chart so that it shows the correct data.

richarddubay committed:
Merge pull request #377 from NewSpring/fix-transaction-date-filter

Fix Giving Graph

richarddubay committed:
fixed email sending

richarddubay committed:
Merge pull request #379 from NewSpring/422-fix-email-sending

Fixed Email Sending

conrad-vanl committed:
Remove wistia api calls (#381)

* remove reliance on wistia api

* Fix video embedUrl loading

* fix snapshots

conrad-vanl committed:
Add conditional for supporting AD login (#382)

IsaacHardy committed:
nmi gateway hotfix

conrad-vanl committed:
use FOLLOWABLE_TOPICS constant if not logged in (#383)

conrad-vanl committed:
Fix recentlyLiked query (#384)

* use aggregator for getting user likes

* fix tests

* fix query for recentlyLiked

* fix test

conrad-vanl committed:
Fix recently liked sort (#386)

* fix recentlyLiked query

*  fix tests

conrad-vanl committed:
Patch colan-less auth tokens (looking at you Edge) (#388)

richarddubay committed:
update to remove the fuse specifics

richarddubay committed:
Merge pull request #389 from NewSpring/church-online

Remove Fuse Church Online
